### ISSUE TITLE
Support for rolling threads

### ISSUE THREAD

**Issue by mahkoh** [ccd0/4chan-x#767]
For some time 4chan has had rolling threads that are used during events such as mod/admin Q&As and currently debate/primary stickies on /pol/.

Rolling threads work as follows: Once the thread reaches a set number of posts (usually 1000), for every new post that is made, the currently oldest post is deleted, thus keeping the number of posts at 1000. This does however not mean that the deleted posts are removed from the DOM, they are simply marked deleted.

Since the threads continuously grow, this causes noticeable slowdown after some time. The normal solution is to refresh the thread, however, since the thread still contains 1000 posts, 4chan-x takes several seconds to apply its DOM modifications after a refresh.

I do not know if the 4chan API allows one to distinguish rolling from normal threads. Depending on this, I see two ways to improve this problem:

1. Add an option to automatically remove deleted posts from the DOM in rolling threads.
2. Add a button to remove all deleted posts from the current thread.

**Comment by ccd0** [comment]
From #790:

> /pol/ has some new thread type which keeps a thread going by deleting its oldest posts as new ones are made. For instance one is up now for Super Tuesday discussion.

> Would be nice if 4chan X had the option to automatically remove posts which are pruned as the thread is updated. The page gets quite laggy as the posts build up and currently it requires a refresh of the page.

**Comment by ccd0** [comment]
Until this is implemented a workaround is to add

   .thread > div:nth-last-of-type(n+1001):not(.opContainer) {
      display: none;
    }

to custom CSS. This will improve performance somewhat, but not as much as removing the posts from the DOM.

**Comment by ccd0** [comment]
I added a pruning tool to the newest beta (v1.11.28.1). Made the limit user-configurable since some people may want to set it even lower than 1000. It's not specific to rolling stickies, either; you can use it in any thread.

![Reply pruning](https://cloud.githubusercontent.com/assets/5894749/13810620/0496f910-eb2e-11e5-8cb9-9029cafb545c.png)

Note: If you're using `Quote Threading`, you'll need to turn that off to use it, since it's not compatible yet.

**Comment by ccd0** [comment]
Note to self: It would be a good idea to show a notice on the page if any replies are hidden.

**Comment by ccd0** [comment]
Also, as of f6010ca298b870684825fb13b3a48747dd38e939, you can have both "Quote Threading" and "Reply Pruning" on at the same time, but you can only choose one of the respective checkboxes in the header menu (`Threading` or `Show Last ___`). But you can switch between them without reloading.

**Comment by ccd0** [comment]
It occurs to me that we can do better than just removing the posts. By tracking the user's scroll position and only removing posts that would be far offscreen, we could handle huge threads without the user noticing anything had changed.

**Comment by mahkoh** [comment]
I'm getting triggered by the background color of posts changing when old posts are purged. Can you please make it so that the background color of each post stays the same?

**Comment by ccd0** [comment]
@mahkoh
Is this with Oneechan / Stylechan?

**Comment by mahkoh** [comment]
Yes, Oneechan has the ability to color even/odd posts differently. Maybe the simplest fix is to only ever purge an even number of posts at a time.
